content/articles/building-scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/index.md

```diff
@@ -0,0 +1,115 @@
+
+```java
+<dependency
+    <groupId>project</groupId
+    <artifactId>notification</artifactId
+    <version>0.0.1-SNAPSHOT</version>
+</dependency>
+```
+![Payment](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/payment.jpg)
+
+## Bonus
+Let us imagine that a `Payment Service` is to be implemented into the Hospital management System later. This is very simple and straightforward. Open the project, right click on the project name; `hospital-project` in this case, click on `project` and then `module`. Enter the details of this new module to conform with the structure we used for the existing modules in the project, install it as a dependency in any or all of the other modules that need its methods and viola, you have a new module in your project. You can try to create a `Logistics Service` to represent the logistic department of the hospital.
+
+![Bonus](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/bonus.jpg)
+
+
+
+## Gaining the benefits of microservices architecture from the multi-module architecture
+To enjoy what microservices offer while using multi-module architecture is simple. This means that you can copy out the jar file of any of the modules/services above – say Payment Service for instance – and deploy it into another entirely different project, for reuse. All you need to do is to write `<packagin>jar</packaging>`in the POM file of the module you want to resuse as jar – ideally, in all the modules. Spring Boot automatically does the rest for you. Now you can reuse that modules anywhere else.
+
+One interesting thing to note is that, if there is an issue in any of the module, you can simply go to that module, fix up the issue and rebuild is as an independent jar, without interfering with the other modules
+
+![Packaging](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/packaging.jpg)
+
+
+
+## Conclusion
+Using the multi-module architecture is more convenient. You have separation of interests and still, a lot of control of the project in one place. Recognising a system downtime and fixing it is faster, and scalability is guaranteed.
+
+This project can be found on the [Github link](https://github.com/teevyne/multi-module-setup)
+
+Thank you and enjoy building.
```

Comment:
Add reference links that might help the reader

Reply:
Oh yes. There are references. I will just add them into the article. Thank you very much